restcomm/restcomm.extension.api/src/main/java/org/restcomm/connect/extension/api/RestcommExtensionGeneric.java

```diff
@@ -43,14 +42,14 @@
      * and either block/allow or modify the session before Restcomm process it
      * @return ExtensionResponse see ExtensionResponse
      */
-    ExtensionResponse preInboundAction(SipServletRequest request);
+    ExtensionResponse preInboundAction(IExtensionRequest extensionRequest);
```

Comment:
i like the idea of being independent to underlying framework (servlet,akka), so good choice on having our own request. It would be still a good idea to be able to recover ref to original framework objects using generic container of objects, so extension impl may use them if required. I noticed in the "init" method we still inject a ServletContext, so the extension is still tied to specific ServletAPI. not sure if is time to have an specific ExtensionContext as well, and pass it as part of the request...

Reply:
ExtensionContext sounds a good idea @jaimecasero. Not sure also if its the proper time to do it. 
I think that since now the team will be more familiar with the Extensions API, for the next task we can plan and estimate enhancements like the one you suggest